compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -524,7 +525,9 @@ class Inliner(call: tpd.Tree, rhs: tpd.Tree)(implicit ctx: Context) {
     }
 
     override def typedSelect(tree: untpd.Select, pt: Type)(implicit ctx: Context): Tree = {
-      val res = super.typedSelect(tree, pt)
+      assert(tree.hasType, tree)
+      val qual1 = typed(tree.qualifier, selectionProto(tree.name, pt, this))
+      val res = untpd.cpy.Select(tree)(qual1, tree.name).withType(tree.typeOpt)
```

Comment:
I'm wondering why use `AnySelectionProto` instead of using `selectionProto(tree.name, pt, this)` as expected type in `Retyper.typedSelect`?